io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/writer/internal/GmlWriterUtil.java

```diff
@@ -129,6 +129,35 @@ public static boolean isID(TypeDefinition type) {
 		}
 	}
 
+	/**
+	 * Determines if the given type is an INSPIRE type
+	 * 
+	 * @param ProdDef the type definition
+	 * @return if the type is an INSPIRE type
+	 */
+	public static boolean isInspireType(PropertyDefinition propDef) {
+		if (propDef.getName().getNamespaceURI().startsWith("http://inspire.ec.europa.eu/schemas/")
+				|| propDef.getName().getNamespaceURI()
+						.startsWith("https://inspire.ec.europa.eu/schemas/")) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Determines if the given parent is a gml:identifier
+	 * 
+	 * @param QName the QName
+	 * @return true if the parent is a gml:identifier
+	 */
+
+	public static boolean isGmlIdentifier(QName QName) {
```

Comment:
I have checked and seen that the value of `gmlNs` for my test dataset is always `http://www.opengis.net/gml/3.2`. What should I check exactly? That it contains `gml`?

Reply:
this `http://www.opengis.net/gml/3.2` itself is gml namespace. 
you might have to check:

```
gmlNs.equals(qName.getNamespaceURI())
```
you can also do a null check for `gmlNs` if it can be null. However, I don't think that it will be `null` 

reference: https://docs.oracle.com/javase/7/docs/api/javax/xml/namespace/QName.html#getNamespaceURI()